### ISSUE TITLE
Where are you putting this loss.py function? Also I am trying to use this to train on my own data but I keep getting empty feature map. Do you know how to train on my own data?

### ISSUE THREAD

**Issue by ishacusp** [CasiaFan/tensorflow_retinanet#17]

**Comment by zongfan2** [comment]
@ishacusp  TF object detection has its own focal loss function integrated into its framework and you could enable it by defining `weighted_sigmoid_focal` in config loss section.  If you'd like to use this customed loss function, modify `object_detection/meta_architecture/ssd_meta_arch` to replace the default classification loss (Line [844](https://github.com/tensorflow/models/blob/master/research/object_detection/meta_architectures/ssd_meta_arch.py#L844)).

**Comment by ishacusp** [comment]
Thanks! I followed your README and used your retinanet.py and feature extractor, to train on my data, but the feature extractor returns an empty map.

**Comment by zongfan2** [comment]
@ishacusp Okay... I get it. This error is caused by inexplicit parameter assignment during retinanet feature extractor initialization when I add `min_level` and `max_level` in base class only. I have fixed this bug and you could pull it down to use. Thanks for reporting.

**Comment by ishacusp** [comment]
Thank You!
